marklogic-data-hub/src/main/java/com/marklogic/hub/ArtifactManager.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2012-2020 MarkLogic Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.marklogic.hub;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+import java.io.IOException;
+import java.nio.file.Path;
+
+/**
+ * Handles writing artifacts to ML server and project directory and reading artifacts from ML server
+ */
+public interface ArtifactManager {
+
+    /**
+     * Gets artifacts of a given type
+     *
+     * @param artifactType - type of artifact the operation is dealing with
+     * @return JsonNode of the artifacts matching a type
+     */
+    public JsonNode getArtifacts(String artifactType);
+
+    /**
+     * Updates an artifact of a given type and name.
+     *
+     * @param artifactType - type of artifact the operation is dealing with
+     * @param artifactName - identifier of the artifact
+     * @param artifactJson - JSON of artifact to persist
+     * @return JsonNode of the artifact matching a type
+     */
+    public JsonNode updateArtifact(String artifactType, String artifactName, JsonNode artifactJson) throws IOException;
+
+    /**
+     * Retrieves an artifact of a given type and name.
+     *
+     * @param artifactType - type of artifact the operation is dealing with
+     * @param artifactName - identifier of the artifact
+     * @return JsonNode of the artifact matching a type and name
+     */
+    public JsonNode getArtifact(String artifactType, String artifactName);
```

Comment:
Can this be strongly typed as an ObjectNode since it should be a single thing?

I mention this for a few other methods. I think the distinction would be useful for clients that aren't just going to be a passthrough. And it's more self-documenting to use ObjectNode or ArrayNode if we know it's going to be one or the other.